src/main/java/seedu/hippocampus/model/person/Birthday.java

```diff
@@ -0,0 +1,68 @@
+package seedu.hippocampus.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.hippocampus.commons.util.AppUtil.checkArgument;
+
+/** Represents a Person's birthday in HippoCampus.
+ * Guarantees: immutable; is always valid.
+ */
+public class Birthday {
+
+    public static final String MESSAGE_CONSTRAINTS = "Birthday must be in the format yyyy-mm-dd";
+
+    public static final String EMPTY_BIRTHDAY_STRING = "";
+    public static final Birthday EMPTY_BIRTHDAY = new Birthday();
+
+    public final String value;
+    private boolean isEmpty = false;
+
+    /**
+     * Constructs a {@code Birthday}.
+     *
+     * @param birthdate A valid birthdate.
+     */
+    public Birthday(String birthdate) {
+        requireNonNull(birthdate);
+        checkArgument(isValidBirthday(birthdate), MESSAGE_CONSTRAINTS);
+        value = birthdate;
+    }
+
+    /**
+     * Constructs an empty birthday.
+     */
+    public Birthday() {
+        value = EMPTY_BIRTHDAY_STRING;
+        isEmpty = true;
+    }
+
+    /**
+     * Returns true if a given birthday is an empty birthday.
+     */
+    public static boolean isEmptyBirthday(Birthday birthday) {
+        return birthday.isEmpty;
+    }
+
+    /**
+     * Returns true if a given string is a valid email.
+     */
+    public static boolean isValidBirthday(String test) {
+        return (test.toCharArray()[4] == '-') && (test.toCharArray()[7] == '-');
```

Comment:
This doesn't seem like a very robust check, since it would allow dates such as 2020-99-99, which would be an invalid birthday as well.

I think we could instead use the Java LocalDate class, which provides the parsing methods as well.

You could refer to what I did in my IP, for the deadlines. https://github.com/nickyfoo/ip/blob/master/src/main/java/duke/command/DeadlineCommand.java

or here for usage of LocalDate class
https://www.baeldung.com/java-string-valid-date

Reply:
That's true. I considered LocalDate but wanted to keep to standardising using isValidBirthday boolean to check for validity. Will try to see if I can use both, else will switch to LocalDate. Thanks!

Reply:
Used LocalDate to check for validity in latest commit.